examples/express-ts/test.js

```diff
@@ -0,0 +1,357 @@
+  });
+
+  it("reads a person's movies", async () => {
+    const output = await req.get(`persons/${jennifer.id}/movies`);
+    const moviesOut = output.data;
+    assert.isArray(moviesOut);
+    assert.equal(moviesOut.length, 2, 'pet count');
+
+    const silverLiningsOut = moviesOut[moviesOut[0].id === silverLinings.id ? 0 : 1];
+    const hungerGamesOut = moviesOut[moviesOut[0].id === hungerGames.id ? 0 : 1];
+
+    assert.deepEqual(silverLiningsOut, silverLinings);
+    assert.deepEqual(hungerGamesOut, hungerGames);
+  });
+});
+
+describe('error handling', () => {
+  it("declines an insert post that's missing a required field", async () => {
+    try {
+      let output = await req.post('persons', {
+        lastName: 'Lawrence',
+        age: 24
+      });
+      assert(false, 'should have errored');
+    } catch (err) {
+      const response = err.response;
+      assert.strictEqual(response.status, 400);
+      const firstNameData = response.data.firstName;
+      assert.exists(firstNameData);
+      assert.strictEqual(firstNameData[0].keyword, 'required');
+    }
+  });
+
+  it("declines an eager insert post that's missing a required field", async () => {
+    try {
+      let output = await req.post(`persons/${jennifer.id}/pets`, {
+        species: 'dog'
+      });
+      assert(false, 'should have errored');
+    } catch (err) {
+      const response = err.response;
+      assert.strictEqual(response.status, 400);
+      const nameData = response.data.name;
+      assert.exists(nameData);
+      assert.strictEqual(nameData[0].keyword, 'required');
+    }
+  });
+
+  it('declines an eager insert post into an invalid model ID', async () => {
+    try {
+      let output = await req.post(`persons/9999/pets`, {
+        name: 'Rex',
+        species: 'dog'
+      });
+      assert(false, 'should have errored');
+    } catch (err) {
+      assert.strictEqual(err.response.status, 404);
+    }
+  });
+});
+
+describe('deleting objects', () => {
+  it('deletes a person', async () => {
+    let output = await req.delete(`persons/${bradley.id}`);
+    assert.strictEqual(output.status, 200);
+    assert(output.data.dropped, 'confirmed drop');
+  });
+});
+
+function checkSubset(actual, expected, updatedAfter = null, createdAfter = null) {
+  if (createdAfter) {
+    assert.exists(actual.createdAt, 'creation time');
+    const createdAt = new Date(actual.createdAt);
+    assert.isAbove(createdAt.getTime(), createdAfter.getTime());
+  }
+  if (updatedAfter) {
+    assert.exists(actual.updatedAt, 'update time');
+    const updatedAt = new Date(actual.updatedAt);
+    assert.isAbove(updatedAt.getTime(), updatedAfter.getTime());
+  }
+  for (const property in expected) {
+    assert.deepEqual(actual[property], expected[property], `property ${property}`);
+  }
+}
+
+function clearDatabase(knex) {
+  const tables = ['Person', 'Movie', 'Animal', 'Person_Movie'];
+  const jobs = [];
+  tables.forEach(table => {
```

Comment:
Do you know about `.map`? This whole method is a one-liner:

```js
return Promise.all(['Person', 'Movie', 'Animal', 'Person_Movie'].map(table => knex(table).del()))
```

Reply:
Yeah, I guess it didn't come to mind. I can do it in one line, but it might be clearer in two:

```js
const tables = ['Person', 'Movie', 'Animal', 'Person_Movie']
return Promise.all(tables.map(table => knex(table).del()))
```

Thanks for the reminder! I still mostly think in Java.